### PR TITLE
fix(processing): restore parameters for WASM tools whose manifest declares none

### PR DESCRIPTION
**Pull request by giswqs**
## The bug

In local (WASM) mode the Processing dialog shows **"This tool has no parameters."** for **138 of the 881** Whitebox tools — every tool under **Hydrology → Flow Routing** (D8 Pointer, D8 Flow Accum, Rho8 Pointer, Dinf Pointer, …), plus Fill Depressions, Aspect, Basins, Sink, Sky View Factor and more. With no parameters rendered there is nothing to configure and nothing to run: Run sends no arguments and the tool exits with `validation error: missing required parameter '…'`.

## Cause

`geolibre-wasm` ships those 138 manifests with an **empty `params` array** while the binary still requires the parameters. Probing every one of the 138 with no arguments produces `validation error: missing required parameter '…'` for **all 138** — so an empty manifest is a metadata gap, not a parameterless tool.

`mergeWasmToolManifests` treated the WASM manifest as authoritative *"even an empty set"*, so that empty array overwrote the catalog snapshot's parameters for every tool present in both sources.

This only ever showed with **Run locally (WASM)** on: the sidecar path already restores catalog parameters for a live tool that reports none, via `mergeCatalogParameterFallbacks` in `ProcessingDialog`.

Not a regression from #1386 — the merged output is byte-identical under geolibre-wasm 0.9.0, 0.10.0, 0.11.0 and 1.0.0 (all four ship the same 138 empty manifests), so the tools have been unconfigurable in WASM mode since the merge was introduced.

## Fix

An **empty** manifest param list falls back to the catalog's parameters, mirroring the sidecar path. A manifest that *declares* params stays authoritative, so #1047 (`reproject_vector`'s `epsg` vs the catalog's `dst_epsg`) and #1073 (scalar-kind corrections for `statement`/`expression`) are untouched.

## Result

All **881** tools now render their parameters — **0** left showing "This tool has no parameters."

Probing the binary with the catalog's parameter names: **129 of 138** accept them as-is. The remaining **9** name their input differently in the binary than in the catalog and still need an upstream manifest fix in `geolibre-rust` (they were unrunnable before this change too):

| tool | catalog input | binary requires |
| --- | --- | --- |
| `average_horizon_distance` | `input` | `dem` |
| `horizon_area` | `input` | `dem` |
| `sky_view_factor` | `input` | `dem` |
| `visibility_index` | `input` | `dem` |
| `distance_to_outlet` | `d8_pointer` | `d8_pntr` |
| `farthest_channel_head` | `d8_pointer` | `d8_pntr` |
| `find_main_stem` | `d8_pointer` | `d8_pntr` |
| `max_branch_length` | `dem` | `input` |
| `minimal_dispersion_flow_algorithm` | `raster` | `input` |

The real upstream fix is for geolibre-rust to emit `params` in these 138 manifests.

## Verification

- Real browser, real data: loaded the source.coop `giswqs/opengeos/dem.tif` COG, opened **Processing → Hydrology → Flow Routing → D8 Pointer** with **Run locally (WASM)** on. It renders `Dem` / `Esri Pntr` / `Output`, runs in the WASM engine and adds **D8 Pointer Output** to the map (`{"path":"/work/d8_pointer_output.tif"}`, status `succeeded`). Fill Depressions, Aspect and Basins render their parameters too.
- Two new unit tests: catalog params are kept when the manifest declares none; a WASM-only tool with no params is still listed (nothing to fall back to).
- 3454 frontend tests pass; `pre-commit` (oxfmt, eslint, npm build) clean.

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed merged local WASM tools with empty `params` being treated as parameterless.
  * When WASM manifest parameter details are missing (empty `params`), the merged result now preserves the catalog’s full parameter definitions and related display metadata.
  * Continued to honor WASM-provided parameter details when present, and kept WASM-only tools with truly empty parameter lists available.
* **Tests**
  * Added coverage for the “empty params” merge edge cases to prevent regressions.

### CODE REVIEW

File: tests/wasm-tool-manifests.test.ts [hunk 1]
```diff
@@ -255,6 +255,47 @@ describe("mergeWasmToolManifests", () => {
...
+      params: [],
+    };
+    const [tool] = mergeWasmToolManifests([catalogD8Pointer], [wasmD8Pointer]);
+    assert.deepEqual(
+      tool.params?.map((param) => param.name),
+      ["dem", "esri_pntr", "output"],
+    );
+    // Still a catalog merge, not a passthrough: display metadata is the catalog's.
+    assert.equal(tool.category, "Hydrology - Flow Routing");
+  });
```
[the remark this answers is not in the document]
Comment (giswqs) [reply tests/wasm-tool-manifests.test.ts:291]: Done in 9c23e430 — the test now asserts `tool.params` deep-equals `catalogD8Pointer.params`, so a fallback that dropped `kind`, `required` or `default` fails it (`kind` decides how the dialog renders each field — a lost `raster_in` turns the layer picker into a text box — and `required`/`default` seed createDefaultValues and gate the Run button).
